pom.xml

```diff
@@ -733,6 +733,34 @@
         <maven.repository.update.freqency>always</maven.repository.update.freqency>
       </properties>
     </profile>
+    <profile> <!-- JENKINS-51247: bypass MNG-6405 by avoiding use of forked executions incl. from source:jar -->
+      <id>fix-superpom-source-fork</id>
+      <activation>
+        <property>
+          <name>performRelease</name>
+          <value>true</value>
+        </property>
+      </activation>
+      <build>
+        <plugins>
+          <plugin>
+            <artifactId>maven-source-plugin</artifactId>
+            <executions>
+              <execution>
+                <id>attach-sources</id>
+                <phase>DO-NOT-ACTUALLY-RUN</phase>
+              </execution>
+              <execution>
+                <id>attach-sources-no-fork</id>
+                <goals>
+                  <goal>jar-no-fork</goal>
```

Comment:
NIT/for my understanding: so the title of the PR is wrong or misleading, shouldn't it be more like:
"Always use `source:jar` for non-release builds, use `source:jar-no-fork` for releases"